### ISSUE TITLE
UndefinedBehavior in repair_reader::make_reader

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#10882]
Saw this locally when testing `materialized_views_test.py::TestMaterializedViews::test_base_replica_repair`

```
DEBUG 2022-06-26 16:19:25,487 [shard 0] repair - is_same_sharding_config: remote_shard=0, remote_shard_count=2, remote_ignore_msb=12
/usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/optional:433:58: runtime error: load of value 190, which is not a valid value for type 'bool'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/optional:433:58 in 
DEBUG 2022-06-26 16:19:25,488 [shard 1] repair - >>> Started Row Level Repair (Follower): local=127.0.92.2, peers=127.0.92.1, repair_meta_id=1, keyspace=ks, cf=t, schema_version=d26508ee-21d9-3eed-9fa8-dbf9ca851810, range=(-inf, -9201723570121857511], seed=2861521008320345211, max_row_buf_siz=33554432
DEBUG 2022-06-26 16:19:25,488 [shard 1] repair - is_same_sharding_config: remote_shard=1, remote_shard_count=2, remote_ignore_msb=12
Aborting on shard 0.
Backtrace:
  0x13b6a150
  0x1b082e3d
  0x1b082a3c
  0x1aee70a4
  0x1af12bfe
  0x1aff3f5c
  0x1aff41bd
  0x1aff3fea
  0x7f9dd5a7aa1f
  /lib64/libc.so.6+0x3d2a1
  /lib64/libc.so.6+0x268a3
  0x13be587e
  0x13be37ab
  0x13bf5afe
  0x13bf7093
  0x13bfaa7a
  0x1891a958
  0x1891a76d
  0x189193cf
  0x18918970
  0x18916072
  0x18900d1c
  0x18a555a6
  0x18a551a8
  0x18a54f13
  0x188cb1b4
  0x188cad84
  0x188caca3
  0x188cac48
  0x188cab2d
  0x18690010
  0x1868ff73
  0x1868ff33
  0x1868fee3
  0x1868fea3
  0x1868fe20
  0x1868fd42
  0x1868fbce
  0x1868f956
  0x1af05d9e
  0x1af0c416
  0x1af11082
  0x1af0ebf1
  0x1accd111
  0x1accac75
  0x13c00ac4
  0x140dd066
  0x140dcf9b
  0x140dce05
  0x13cc084d
  0x13bff2a1
  /lib64/libc.so.6+0x27b74
  0x13b1e22d
```

```
__sanitizer::Abort() at crtstuff.c:?
__sanitizer::Die() at crtstuff.c:?
__ubsan::ScopedReport::~ScopedReport() at crtstuff.c:?
handleLoadInvalidValue(__ubsan::InvalidValueData*, unsigned long, __ubsan::ReportOptions) at crtstuff.c:?
__ubsan_handle_load_invalid_value at ??:?
std::_Optional_base_impl<evictable_reader_handle_v2, std::_Optional_base<evictable_reader_handle_v2, true, true> >::_M_is_engaged() const at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/optional:433
std::enable_if<__and_v<std::__not_<std::is_same<std::optional<evictable_reader_handle_v2>, std::remove_cv<std::remove_reference<evictable_reader_handle_v2>::type>::type> >, std::__not_<std::__and_<std::is_scalar<evictable_reader_handle_v2>, std::is_same<evictable_reader_handle_v2, std::decay<evictable_reader_handle_v2>::type> > >, std::is_constructible<evictable_reader_handle_v2, evictable_reader_handle_v2>, std::is_assignable<evictable_reader_handle_v2&, evictable_reader_handle_v2> >, std::optional<evictable_reader_handle_v2>&>::type std::optional<evictable_reader_handle_v2>::operator=<evictable_reader_handle_v2>(evictable_reader_handle_v2&&) at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/optional:807
std::enable_if<(__assignable<flat_mutation_reader_v2, evictable_reader_handle_v2>)(), std::tuple<flat_mutation_reader_v2&, std::optional<evictable_reader_handle_v2>&>&>::type std::tuple<flat_mutation_reader_v2&, std::optional<evictable_reader_handle_v2>&>::operator=<flat_mutation_reader_v2, evictable_reader_handle_v2>(std::pair<flat_mutation_reader_v2, evictable_reader_handle_v2>&&) at /usr/lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/tuple:1329
repair_reader::make_reader(seastar::sharded<replica::database>&, replica::table&, seastar::bool_class<is_local_reader_tag>) at ./repair/row_level.cc:298
repair_reader at ./repair/row_level.cc:340
repair_meta at ./repair/row_level.cc:815
```

**Comment by bhalevy** [comment]
Introduced in 080f00b99a3d944f8d0b4e6a6b4d1e5f0f292777
(merged in 4.2, backported to 4.1)

**Comment by avikivity** [comment]
Backported to 5.0 (5.1 already had it)
